src/read.c

```diff
@@ -950,7 +950,7 @@ static avifResult avifDecoderItemMaxExtent(const avifDecoderItem * item, const a
                     return AVIF_RESULT_BMFF_PARSE_FAILED;
                 }
                 startOffset += remainingOffset;
-                extentSize -= remainingOffset;
+                extentSize = (size_t)((uint64_t)extentSize - remainingOffset);
```

Comment:
You may not need to do the intermediate `(uint64_t)` cast. `(extentSize - remainingOffset)` should be promoted to the biggest of `sizeof(extentSize)` and `sizeof(remainingOffset)`.

Reply:
Thanks for the pull request.

Because of the `remainingOffset >= extentSize` check at line 945, we know `remainingOffset < extentSize` here. So an alternative fix, which requires only one cast, is the following:

```
                extentSize -=(size_t)remainingOffset;
```

This cast is safe because `remainingOffset < extentSize`  and `extentSize` is a `size_t`.

Reply:
Christoph: I made this change in the second commit.